### PR TITLE
{lang}[intel/2015a] R 3.1.2 + deps (OK)

### PR DESCRIPTION
**Pull request by boegel**
depends on #1288 for zlib, libreadline, ncurses easyconfigs

cc @hajgato, @hpcleuven, @sbecuwe

removed deprecated/disappeared extensions: Defaults, odesolve, Rniftilib, igraph0, acepack, Snowball, gdsfmt, SNPRelate
removed because doesn't build: forecast (needs patch?), Mcomp (depends on forecast)
extensions moved to dedicated easyconfig: XML, RCurl, rtracklayer, BSgenome (#1292)

### CODE REVIEW

**Comment by boegel** [comment]
`BSGenome` package is not included, since it depends on the `RCurl` package now, which in turn requires `cURL` as a dependency

since `cURL` is not required as a dependency for Python, the cleanest way is to install `BSGenome` as a stand-alone module instead (PR in the works for that)

File: easybuild/easyconfigs/r/R/R-3.1.2-intel-2015a.eb [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    ('RWekajars', '3.7.11-1', ext_options),
+    ('RWeka', '0.4-23', ext_options),
+    ('slam', '0.1-32', ext_options),
+    ('spacodiR', '0.13.0115', ext_options),
+    ('tm', '0.6', ext_options),
+    ('TraMineR', '1.8-8', ext_options),
+    ('untb', '1.7-2', ext_options),
+    ('chemometrics', '1.3.8', ext_options),
+    ('FNN', '1.1', ext_options),
+    #('forecast', '5.8', ext_options),  # doesn't build
```
Comment (boegel) [inline easybuild/easyconfigs/r/R/R-3.1.2-intel-2015a.eb]: @hajgato: `forecast` doesn't build, see error message mentioned below... this probably requires a patch of some sort

Is it worth the trouble, or can we just exclude `forecast` and `Mcomp` (which depends on it)?

**Comment by boegel** [comment]
@wpoely86: please review

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/bb55107c0fb24ac4da20 for a full test report.

File: easybuild/easyconfigs/l/libpng/libpng-1.6.16-intel-2015a.eb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+name = 'libpng'
+version = '1.6.16'
+
+homepage = 'http://www.libpng.org/pub/png/libpng.html'
+description = "libpng is the official PNG reference library"
+
+toolchain = {'name': 'intel', 'version': '2015a'}
+toolchainopts = {'pic': True}
+
+source_urls = [SOURCEFORGE_SOURCE]
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libpng/libpng-1.6.16-intel-2015a.eb]: maybe put this consistently in the same order everywhere?

**Comment by wpoely86** [comment]
looks fine

File: easybuild/easyconfigs/l/libpng/libpng-1.6.16-intel-2015a.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/l/libpng/libpng-1.6.16-intel-2015a.eb]: I will look into that at some point, but not without a script that allows us to enforce this consistently...

File: easybuild/easyconfigs/r/R/R-3.1.2-intel-2015a.eb (hunk 1 above)
Comment (hajgato) [inline easybuild/easyconfigs/r/R/R-3.1.2-intel-2015a.eb]: @boegel: I needed those for something, but I dont remember for what. I can make it later.

Comment (boegel) [inline easybuild/easyconfigs/r/R/R-3.1.2-intel-2015a.eb]: we can re-add these later when we figure out how to fix their build, I'll remove them for now

**Comment by boegel** [comment]
@pescobar: you have an updated R easyconfig too?

**Comment by boegel** [comment]
@pescobar's easyconfig is available at http://paste.linuxmint.com/view/8fiy/, I'll compare it with what I have here, and try to include any missing packages

**Comment by boegel** [comment]
see also #1220...

**Comment by boegel** [comment]
I've merged @pescobar's version of the R v3.1.2 easyconfig into mine, retesting it now...

**Comment by boegel** [comment]
added more extensions, as suggested by @pescobar 

I had to exclude a bunch of his suggestions though, because the extensions require extra dependencies (cfr. #1292):

```
    ('biomaRt', '2.22.0', bioconductor_options),  # requires XML, RCurl
    ('annotate', '1.44.0', bioconductor_options),  # requires XML
    ('GSEABase', '1.28.0', bioconductor_options),  # requires XML, annotate
    ('genefilter', '1.48.1', bioconductor_options),  # requires annotate
    ('Category', '2.32.0', bioconductor_options),  # requires GSEABase, genefilter, annotate
    ('GOstats', '2.32.0', bioconductor_options),  # requires annotate, Category
    ('geneplotter', '1.44.0', bioconductor_options),  # requires annotate
    ('DESeq2', '1.6.3', bioconductor_options),  # requires geneplotter, genefilter
    ('GenomicFeatures', '1.18.3', bioconductor_options),  # requires rtracklayer, biomaRt, RCurl
    ('VariantAnnotation', '1.12.8', bioconductor_options),  # requires BSgenome, rtracklayer, GenomicFeatures
    ('biovizBase', '1.14.1', bioconductor_options),  # requires GenomicFeatures, VariantAnnotation
    ('OrganismDbi', '1.8.0', bioconductor_options),  # requires GenomicFeatures
    ('ggbio', '1.14.0', bioconductor_options),  # biovizBase, BSgenome, VariantAnnotation, rtracklayer, GenomicFeatures, OrganismDbi
    ('ReportingTools', '2.6.0', bioconductor_options),  # requires Category, GOstats, annotate, GSEABase, XML, DESeq2, ggbio
    ('affycoretools', '1.38.0', bioconductor_options),  # requires biomaRt, GOstats, annotate, genefilter, ReportingTool
    ('AgiMicroRna', '2.16.0', bioconductor_options),  # requires affycoretools
    ('DESeq', '1.18.0', bioconductor_options),  # requires genefilter, geneplotter
    ('GenomeGraphs', '1.26.0', bioconductor_options),  # requires biomaRt
    ('geneLenDataBase', '1.1.1', bioconductor_options),  # requires rtracklayer, GenomicFeatures
    ('goseq', '1.18.0', bioconductor_options),  # requires geneLenDataBase
    ('TxDb.Hsapiens.UCSC.hg19.knownGene', '3.0.0', bioconductor_options),  # requires GenomicFeatures
    ('Homo.sapiens', '1.1.2', bioconductor_options),  # requires OrganismDbi, GenomicFeatures, TxDb.Hsapiens.UCSC.hg19.knownGene
    ('KEGGgraph', '1.24.0', bioconductor_options),  # requires XML
    ('httr', '0.6.1', ext_options),  # requires RCurl
    ('KEGGREST', '1.6.2', bioconductor_options),  # requires httr
    ('KEGGprofile', '1.8.0', bioconductor_options),  # requires XML, KEGGREST, biomaRt
    ('minfi', '1.12.0', bioconductor_options),  # requires genefilter
    ('methylumi', '2.12.0', bioconductor_options),  # requires minfi, annotate, genefilter
    ('lumi', '2.18.0', bioconductor_options),  # requires methylumi, GenomicFeatures, annotate
    ('Gviz', '1.10.5', bioconductor_options),  # requires rtracklayer, biomaRt, GenomicFeatures, BSgenome, biovizBase
    ('methyAnalysis', '1.8.0', bioconductor_options),  # requires lumi, methylumi, Gviz, rtracklayer, GenomicFeatures, annotate, genefilter, biomaRt
    ('pathview', '1.6.0', bioconductor_options),  # requires KEGGgraph, KEGGREST
    ('SPIA', '2.18.0', bioconductor_options),  # requires KEGGgraph
    ('cummeRbund', '2.8.2', bioconductor_options),  # requires rtracklayer, Gviz
    ('spliceR', '1.8.0', bioconductor_options),  # requires cummeRbund, rtracklayer
```


We need a better solution for this, to 'pack' all these extensions in a single easyconfig, rather than having to create separate easyconfigs for each of these, which is madness.

@wpoely86: up for testing this as a whole?

**Comment by boegel** [comment]
@rjeschmi: this should be ready to test with `foss`, let me know if you're up for that...

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 5 out of 8 (6 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/816ee5ded29623f8644f for a full test report.

**Comment by boegel** [comment]
@wpoely86: requires a manual download for Java...

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 6 out of 8 (6 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/1832ab952f3e31fd57fe for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/34a13926f9dac31b59eb for a full test report.

**Comment by boegel** [comment]
@wpoely86: hmm, Lmod is broken when the R module is being loaded because it provides `libreadline.so.6`, see:

```
 /usr/bin/lua: symbol lookup error: /home/ward/tmp/eb_sandbox_20150121_13-28-50_wcR0E/software/libreadline/6.3-intel-2015a/lib/libreadline.so.6
```

**Comment by wpoely86** [comment]
@boegel I'm aware of the issue. On Debian, `libreadline` should link to `libtinfo` but it doesn't happen for some reason. My current solution is to force it manually.

**Comment by boegel** [comment]
@wpoely86: to force what manually, exactly? is there any way in which you can test this PR?

**Comment by pescobar** [comment]
it looks fine for me. +1 for adding the extra download_urls for the bioconductor packages...

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/b0cdd8df4bca01c0bf56 for a full test report.

**Comment by boegel** [comment]
@wpoely86: thanks a lot for testing this!

Since this works on two pretty different types of systems (SL vs Debian), I'm merging this in.

cc: @sbecuwe, @hpcleuven, @gjbex, @hajgato

**Comment by rjeschmi** [comment]
Test report by @rjeschmi
**FAILED**
Build succeeded for 0 out of 2 (1 easyconfigs in this PR)
Linux centos 6.6, Intel(R) Xeon(R) CPU           E5440  @ 2.83GHz, Python 2.6.6
See https://gist.github.com/a373a1d165c910d00ac8 for a full test report.

**Comment by rjeschmi** [comment]
sorry, the report isn't really important here. I'll open another PR when I have a foss config

**Comment by boegel** [comment]
@rjeschmi: I disagree, I think it would be useful, but feel free to redirect it to a PR with a `foss` easyconfig included in it. :)

**Pull request merged**